scraper_cleveland.py

```diff
@@ -0,0 +1,310 @@
+    """
+    The function scrapes a specific attribute from a details page
+    :param driver: the webdriver in use
+    :param detail: the attribute needed to scrape
+    :param xpath: the xpath of the attribute needed to scrape
+    :return: returns a dictionary key-value pair of the attribute to scrape
+    """
+    obj = dict()
+    try:
+        element = driver.find_element_by_xpath(xpath)
+    except:
+        pass
+    else:
+        val = basic_clean(element.text)
+        if detail == DetailedHeader.CONSIDERATION.value:  # for `consideration` attribute, convert to float number
+            try:
+                val = float(val.replace('$', '').replace(',', ''))
+            except ValueError:
+                val = 0
+        obj[detail] = val
+    return obj
+
+
+def scrape_details(driver: webdriver, url) -> dict:
+    """
+    The function scrapes all needed attributes off the details page of each permit
+    :param driver: the webdriver in use
+    :param url: the url of the details page
+    :return: return a JSON-like object contains the attributes get from the detail page
+    """
+    obj = dict()
+    driver.execute_script("window.open('');")  # open a new window
+    driver.switch_to.window(driver.window_handles[1])
+    driver.get(url)  # open the details page
+    collapses = wait_for_elements_load(driver, By.XPATH, Element.UNFOLD_BUTTON_XPATH.value)  # unfold all sub levels
+    for button in collapses:
+        button.click()
+        time.sleep(1)  # wait for hidden content reveals
+    details = list(DETAILS_MAP.keys())  # get the attributes and the referring content locator from a dict constant
+    for detail in details:
+        xpath = DETAILS_MAP[detail]
+        for path in xpath:
+            obj.update(get_detail(driver, detail, path))
+    # time.sleep(3)
+    driver.close()
+    return obj
+
+
+def scrape_content(driver: webdriver, headers: list, content_rows: list) -> list:
+    """
+    Get the content of the table
+    :param driver: the webdriver in use
+    :param headers: table header
+    :param content_rows: row elements of the table content, usually presented as `tr` tags
+    :return: returns a list of permit object in dictionary type
+    """
+    objs = []
+    obj_detail = dict()
+    for row in content_rows:
+        cells = row.find_elements_by_tag_name('span')
+        vals = []
+        for cell in cells:
+            val = basic_clean(cell.text)
+            vals.append(val)
+        url_cell = row.find_elements_by_tag_name('td')[TableHTML.URL_CELL_POSITION.value]
+        try:
+            url = url_cell.find_element_by_tag_name('a').get_attribute('href')  # get the url for the detail page
+        except:
+            pass  # pass to the next permit if details page is not available
+        else:
+            obj_detail = scrape_details(driver, url)  # scrape off the detail page
+            driver.switch_to_window(driver.window_handles[0])  # get back to the table list
```

Comment:
Based on the code I don't see the needs to go back table list window. Since you already have all `urls` in the `content_rows` object

Reply:
The `scrape_detail()` process is running every cycle in the for loop, row by row. Each cycle will get only 1 `url`. If not switching back to the table, it wont get the next `url`.

Reply:
The `url` is from `url_cell` which is from `row` which is just one element in `content_rows` for each iteration. And for each iteration, you don't need to access the table again to get the row, right?